### DISCUSSION TITLE
TfsCmdlets Get-TFSGroup returns  Error getting descriptor for project '' API resource location '' is not registered on

### DISCUSSION THREAD

**Discussion by rhino2334** [igoravl/TfsCmdlets#233] [Q&A] [answered]
Are all of the functions of TfsCmdlets compatible with azure devops server 2020?
I'm able to use Get-TfsTeamProject to get the list of all projects within a project collection, but when i try to loop through and then use a project name with Get-TfsGroup as shown in the following line, `"Get-TfsGroup -Collection $FISCollection -Project $proj.Name -Scope Project`", I get this error:  
Get-TfsGroup: Error getting descriptor for project '_ProjectName_' (API resource location _GUID_ is not registered on _https://myAzureDevopsUrl/CollectionName_.. )

**Comment by igoravl** [comment, marked as answer]
You're right - Azure DevOps Server 2020 does not have the API required for the `Get-TfsGroup` cmdlet.

As an workaround, you can try the following to get a single group:

```
$proj = 'YOUR_TEAM_PROJECT_NAME_HERE'
$group = 'YOUR_GROUP_NAME_HERE'
$g = Get-TfsIdentity "[$proj]\$group"
$members = $g.MemberIds | ForEach-Object { Get-TfsIdentity $_ }
```

However, if you're trying to _enumerate_ the groups of said project, then there's another workaround. The basic idea here is to via the members of the *Project Valid Users* group. It contains all valid users (and _groups_) in the given team project. From there, you can get the groups you're looking for:

```
$proj = 'YOUR_TEAM_PROJECT_NAME_HERE'
$pvu = Get-TfsIdentity "[$proj]\Project Valid Users"
$allIdentities = $pvu.MemberIds | ForEach-Object { Get-TfsIdentity $_ }
$groups = $allIdentities | Where-Object IsContainer -eq $true
```

HTH!

I'm adding a fix to it to the backlog, so that Azure DevOps Server users can user Get-TfsGroup in the future.

**Comment by rhino2334** [comment]
Thanks for the quick response Igor!  I will try your suggested workarounds.

**Reply by rhino2334** [reply]
The workaround worked perfectly.  Thanks again.
